### ISSUE TITLE
Stripped libraries break valgrind

### ISSUE THREAD

**Issue by cstrouse** [skycocker/chromebrew#746]
@jam7 @lyxell Any ideas for what to do here since we're shipping stripped binaries and libraries now? Noticed this when updating the package (#747)
 
```bash
==32421== Memcheck, a memory error detector
==32421== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==32421== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==32421== Command: /bin/ls
==32421== 

valgrind:  Fatal error at startup: a function redirection
valgrind:  which is mandatory for this platform-tool combination
valgrind:  cannot be set up.  Details of the redirection are:
valgrind:  
valgrind:  A must-be-redirected function
valgrind:  whose name matches the pattern:      strlen
valgrind:  in an object with soname matching:   ld-linux-x86-64.so.2
valgrind:  was not found whilst processing
valgrind:  symbols from the object with soname: ld-linux-x86-64.so.2
valgrind:  
valgrind:  Possible fixes: (1, short term): install glibc's debuginfo
valgrind:  package on this machine.  (2, longer term): ask the packagers
valgrind:  for your Linux distribution to please in future ship a non-
valgrind:  stripped ld.so (or whatever the dynamic linker .so is called)
valgrind:  that exports the above-named function using the standard
valgrind:  calling conventions for this platform.  The package you need
valgrind:  to install for fix (1) is called
valgrind:  
valgrind:    On Debian, Ubuntu:                 libc6-dbg
valgrind:    On SuSE, openSuSE, Fedora, RHEL:   glibc-debuginfo
valgrind:  
valgrind:  Note that if you are debugging a 32 bit process on a
valgrind:  64 bit system, you will need a corresponding 32 bit debuginfo
valgrind:  package (e.g. libc6-dbg:i386).
valgrind:  
valgrind:  Cannot continue -- exiting now.  Sorry.
```

**Comment by jam7** [comment]
Is it really caused by ld-linux-x86-64.so.2?

Chromebrew doesn't provide it.  `/usr/local/lib64/ld-linux-x86-64.so.2` is a symbolic link to `/lib64/ld-linux-x86-64.so.2`.  If it is caused by a part of glibc, it is possible to distribute debuginfo or something.

For example, can you try old toolschains to see what happens with valgrind?

**Comment by jam7** [comment]
I tested it with library files not stripped, but the result is same.  I even installed not stripped `ld-linux-x86-64.so.2` instead of making symbolic links, but the result is same.  Please let me know if you know how to fix it.  Thanks.

**Comment by yancouto** [comment]
I had this same problem and fixed it by applying [this patch](https://bugsfiles.kde.org/attachment.cgi?id=82043) before compiling (found it [here](https://bugs.kde.org/show_bug.cgi?id=286864)).

Should this procedure be added to `valgrind.rb`? If so, I could do that.

**Comment by jam7** [comment]
Oops, this issue has been closed automatically when I merge #959.

Anyway, #959 fixes this problem.  Please be sure to set CREW_NOT_STRIP environment variable to 1 when you install target library and binary for valgrind to not strip them.
